runtime/lua/vim/lsp.lua

```diff
@@ -588,6 +588,12 @@ function lsp.start_client(config)
       -- process is not alive then the server should exit (see exit notification)
       -- its process.
       processId = uv.getpid();
+      -- Information about the client
+      -- since 3.15.0
+      clientInfo = {
+        name = "Neovim",
+        version = vim.fn.matchstr(vim.fn.execute("version"), [[NVIM \zs[^\n]\+]]) or vim.Nil
```

Comment:
it's a bit sad to run a regex to get a short version, `api_info()` returns a dict which is too heavy. We could expose some of the variables from version.{h,c} to lua to make it easier  (could be added to `nlua_state_init`)

Reply:
For now would you rather me just remove the `version`, since it is optional?

Reply:
Is vim.fn.api_info().version.minor not sufficient?

Reply:
Actually maybe that would be better than nothing. It would just show `5`. Do you think that's preferable? If so I can change it to that.

Reply:
I think I see what teto is saying, I'll make the c binding

Reply:
Switched this.

Reply:
If no one beats me to it, I can add this today or tomorrow